### ISSUE TITLE
Training Explanation Please.

### ISSUE THREAD

**Issue by tjc808** [alexjc/neural-enhance#179]
Hello,

Could someone help me make sense of train proccess and awnser a few questions about the model I am traing now? I hope this will help other new comers as well.

#1
(pyvenv) dev@devops:~/neural-enhance$ ./enhance.py --train "mydataset/*.jpg" --model custom --train-scales=2 --epochs=50 --perceptual-layer=conv2_2 --smoothness-weight=1e7 --adversary-weight=0.0 --generator-blocks=4 --generator-filters=64 --device=cpu

This is my command to train my model for super resolution. Could someone explain each of the optional argument(--), in better detail than the read me?

--train "mydataset/*.jpg" # Location of my training data.
--model custom # README expliantion good.
--train-scales=2 #
--epochs=50 #
--perceptual-layer=conv2_2 #
--smoothness-weight=1e7 #
--adversary-weight=0.0 #
--generator-blocks=4 #
--generator-filters=64 #
--device=cpu # README expliantion good.

Also any recommendations on the setting for the following two optional argument for training?
--model=denoise	
--model=deblur
#2

#3
Please explain the output of my ./enhance.py excuted command. Please break one # down with and explination for each line. Based on my the Epoch am I making progress with producing a use able model? At #10 I see the training produced model file, first why and second would I have to modify anything to use the newly generated model during image enhancment attempts? 

Epoch #1 at 4757.6s, lr=1.00e-04
  - generator total=2.44e+05 prcpt=2.42e+05 smthn=1.53e+03 advrs=0.00e+00
  - discriminator -0.306622 0 0.306622 0
Epoch #2 at 9525.9s, lr=1.00e-04
  - generator total=1.86e+05 prcpt=1.83e+05 smthn=3.77e+03 advrs=0.00e+00
  - discriminator 0.719297 15 -0.719297 15
  - generator now optimizing against discriminator.
Epoch #3 at 14308.8s, lr=1.00e-04
  - generator total=1.62e+05 prcpt=1.58e+05 smthn=4.55e+03 advrs=0.00e+00
  - discriminator 0.828149 15 -0.828149 15
Epoch #4 at 19134.2s, lr=1.00e-04
  - generator total=1.55e+05 prcpt=1.51e+05 smthn=4.88e+03 advrs=0.00e+00
  - discriminator 0.882796 15 -0.882796 15
Epoch #5 at 23973.5s, lr=1.00e-04
  - generator total=1.51e+05 prcpt=1.46e+05 smthn=5.14e+03 advrs=0.00e+00
  - discriminator 0.885059 15 -0.885059 15
Epoch #6 at 28817.5s, lr=1.00e-04
  - generator total=1.49e+05 prcpt=1.44e+05 smthn=4.92e+03 advrs=0.00e+00
  - discriminator 0.911471 15 -0.911471 15
Epoch #7 at 33653.6s, lr=1.00e-04
  - generator total=1.47e+05 prcpt=1.42e+05 smthn=5.03e+03 advrs=0.00e+00
  - discriminator 0.930406 15 -0.930406 15
Epoch #8 at 38496.1s, lr=1.00e-04
  - generator total=1.46e+05 prcpt=1.40e+05 smthn=5.27e+03 advrs=0.00e+00
  - discriminator 0.947019 15 -0.947019 15
Epoch #9 at 43333.5s, lr=1.00e-04
  - generator total=1.44e+05 prcpt=1.39e+05 smthn=5.08e+03 advrs=0.00e+00
  - discriminator 0.950581 15 -0.950581 15
Epoch #10 at 48233.0s, lr=1.00e-04
  - generator total=1.41e+05 prcpt=1.36e+05 smthn=5.07e+03 advrs=0.00e+00
  - discriminator 0.958006 15 -0.958006 15
  - saving current generator layers to disk...
  - Saved model as `ne4x-photo-custom-0.3.pkl.bz2` after training.
Epoch #11 at 53288.5s, lr=1.00e-04
  - generator total=1.43e+05 prcpt=1.38e+05 smthn=5.19e+03 advrs=0.00e+00
  - discriminator 0.965029 15 -0.965029 15
Epoch #12 at 58185.5s, lr=1.00e-04
  - generator total=1.40e+05 prcpt=1.35e+05 smthn=5.17e+03 advrs=0.00e+00
  - discriminator 0.971206 15 -0.971206 15
Epoch #13 at 62992.7s, lr=1.00e-04
  - generator total=1.44e+05 prcpt=1.39e+05 smthn=5.28e+03 advrs=0.00e+00
  - discriminator 0.976212 15 -0.976212 15
Epoch #14 at 67794.8s, lr=1.00e-04
  - generator total=1.40e+05 prcpt=1.35e+05 smthn=5.27e+03 advrs=0.00e+00
  - discriminator 0.97688 15 -0.97688 15
Epoch #15 at 72610.9s, lr=1.00e-04
  - generator total=1.42e+05 prcpt=1.37e+05 smthn=5.51e+03 advrs=0.00e+00
  - discriminator 0.962631 15 -0.962631 15

**Comment by 5OPH1A** [comment]
agreed, can we have further explanation on how to train the model using our own images? Thanks in advance.
